common/src/main/java/io/netty/util/NettyRuntime.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.util;
+
+import io.netty.util.internal.SystemPropertyUtil;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A utility class for wrapping calls to {@link Runtime}.
+ */
+public final class NettyRuntime {
+
+    private static final AtomicInteger availableProcessors = new AtomicInteger();
```

Comment:
> What do you use to ensure correct class loading order?

It's simple: do not touch any Netty classes before invoking `NettyRuntime#setAvailableProcessors`. We do the same thing [today](https://github.com/elastic/elasticsearch/blob/2766b08ff45a039d9d7b15435d4680f915f4b1e1/modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Utils.java#L43-L52) with setting the logger factory. This will not be able to be in a static block anymore since it will require input from the user configuration, but I've already tested an approach that works.

> Anyway other netty users may failing on this and will be forced delve into the details of the internal implementation.

Most users will not care, and for those that do, I think that the Javadocs included with this PR are clear on the expectations.

> IMHO it's not clear and SOLID-like way...

Sorry, I don't understand this at all.